doc/deployment/deploy_custom.md

```diff
@@ -1,6 +1,118 @@
+
+This document covers using `pachctl deploy custom ... --dry-run` to create a manifest for a custom, on-premises deployment.
+We won't directly address automating your deployment by means of customization scripts,
+but we do encourage you to treat your [infrastructure as code](./on-premises.html#infrastructure-as-code).
+
+## Anatomy of a Pachyderm deployment manifest
+
+When you run the `pachctl deploy ...` command with the `--dry-run` flag,
+you're generating a JSON-encoded Kubernetes manifest in one stream to standard output. 
+That manifest itself consists of a number of smaller manifests,
+corresponding to a particular aspect of a Pachyderm deployment.
+
+Pachyderm has three sets of application components usually deployed:
+- `pachd`, the main Pachyderm pod
+- `etcd`, the administrative datastore for `pachd`
+- `dash`, the web-based enterprise ui for Pachyderm
+We'll explore all of these within this document.
+
+In general, there are two categories of manifests in the file,
+roles-and-permissions-related and application-related.
+
+## Roles-and-permissions-related manifests
+
+### ServiceAccount
+
+Usually at the top of the file, this manifest has the `kind` key set to `ServiceAccount`. 
+[ServiceAccounts](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) are a way Kubernetes can assign namespace-specific privileges to applications in a lightweight way.
+Pachyderm's service account is called, appropriately enough, `pachyderm`.
+
+### Role or ClusterRole
+
+The next manifest will usually be of `kind` `Role` or `ClusterRole`, 
+depending on whether you  used the `--local-roles` flag `pachctl deploy` command.
+
+### RoleBinding or ClusterRoleBinding
+
+This manifest binds the `Rule` or `ClusterRole` to the `ServiceAccount` created above.
+
+## Application-related
+
+### PersistentVolume
+
+If you don't use [StatefulSets](./on_premises.html#statefulsets) to deploy Pachyderm
+(that is, you don't specify `--dynamic-etcd-nodes` flag), 
+the value you specify for `--persistent-disk` will cause `pachctl` to write a manifest for creating a [`PersistentVolume`](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) for Pachyderm's `etcd` to use in its [`PersistentVolumeClaim`](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims).
+
+### PersistentVolumeClaim
+
+This is the `PersistentVolumeClaim` that Pachyderm's `etcd` will use if you don't deploy using [StatefulSets](./on_premises.html#statefulsets).
+You'll see this manifest's name in the `etcd` Deployment manifest, below.
+
+### StorageClass
+
+If you *do* use [StatefulSets](./on_premises.html#statefulsets) to deploy Pachyderm
+(that is, you use `--dynamic-etcd-nodes` flag), 
+this manifest, if needed, will specify the kind of storage and provisioner that's appropriate for what you've specified in the `--persistent-disk` flag. 
```

Comment:
why is "if needed" there?

Reply:
When deploying to Azure, we don't generate this manifest. The provisioner appears to be hardwired in some way, so it doesn't need to be specified. It's unclear to me if other versions of Kubernetes could have that same behavior.